go/vt/vttablet/onlineddl/executor.go

```diff
@@ -3246,17 +3262,22 @@ func (e *Executor) reviewRunningMigrations(ctx context.Context) (countRunnning i
 				// understand whether "now is a good time" or "not there yet"
 				_ = e.updateMigrationReadyToComplete(ctx, uuid, isReady)
 				if !isReady {
-					return nil
-				}
-				if postponeCompletion {
-					// override. Even if migration is ready, we do not complete it.
+					// The migration is not ready yet.
 					return nil
 				}
 				if strategySetting.IsInOrderCompletion() {
+					var pendingMigrationsCount uint64
 					if len(pendingMigrationsUUIDs) > 0 && pendingMigrationsUUIDs[0] != onlineDDL.UUID {
-						// wait for earlier pending migrations to complete
-						return nil
+						pendingMigrationsCount = getInOrderCompletionPendingCount(onlineDDL, pendingMigrationsUUIDs)
+						postponeCompletion = true
 					}
+					if err = e.updatePostponedByInOrderCompletions(ctx, onlineDDL.UUID, pendingMigrationsCount); err != nil {
```

Comment:
Update the new count field if we are waiting, or if we find we are no longer waiting (`0`)